aws/adapter.go

```diff
@@ -399,3 +476,102 @@ func getTag(tags map[string]string, tagName string) (string, error) {
+			a.obsoleteInstances = append(a.obsoleteInstances, instanceID)
+		} else {
+			// Instance exists, so keep it in the list of single instances
+			newSingleInstances[instanceID] = details
+		}
+	}
+	a.singleInstances = newSingleInstances
+
+	// update ASGs (create new map to get rid of deleted ASGs)
+	newAutoScalingGroups := make(map[string]*autoScalingGroupDetails)
+	autoScalingGroupsToFetchMap := make(map[string]bool)
+	for instanceID, details := range a.ec2Details {
+		asgName, err := getAutoScalingGroupName(details.tags)
+		if err != nil {
+			// Instance is not in ASG, save in single instances list.
+			a.singleInstances[instanceID] = details
+			continue
+		}
+		if _, ok := newAutoScalingGroups[asgName]; !ok {
+			if _, ok := a.autoScalingGroups[asgName]; ok {
+				newAutoScalingGroups[asgName] = a.autoScalingGroups[asgName]
+			} else {
+				// Save ASGs that have to be loaded to load all of them in one API call
+				autoScalingGroupsToFetchMap[asgName] = true
+			}
+		}
+	}
+
+	autoScalingGroupsToFetch := make([]string, 0, len(autoScalingGroupsToFetchMap))
+	for asgName := range autoScalingGroupsToFetchMap {
+		autoScalingGroupsToFetch = append(autoScalingGroupsToFetch, asgName)
+	}
+
+	if len(autoScalingGroupsToFetch) != 0 {
+		fetchedAutoScalingGroups, err := getAutoScalingGroupsByName(a.autoscaling, autoScalingGroupsToFetch)
+		if err != nil {
+			log.Printf("failed fetching Auto Scaling Groups details: %v", err)
+		} else {
+			for name, asg := range fetchedAutoScalingGroups {
+				newAutoScalingGroups[name] = asg
+			}
+		}
+	}
+
+	a.autoScalingGroups = newAutoScalingGroups
+	return nil
+}
+
+// Create EC2 filter that will be used to filter instances when calling DescribeInstances
+// later on each cycle. Filter is based on value of customTagFilterEnvVarName environment
+// veriable. If it is undefined or could not be parsed, default filter is returned which
+// filters on kubernetesClusterTag tag value and kubernetesNodeRoleTag existance.
+func parseFilters(clusterId string) []*ec2.Filter {
+	if filter, ok := os.LookupEnv(customTagFilterEnvVarName); ok {
+		terms := strings.Fields(filter)
+		filters := make([]*ec2.Filter, len(terms))
+		for i, term := range terms {
+			parts := strings.Split(term, "=")
+			if len(parts) != 2 {
+				log.Printf("failed parsing %s, falling back to default", customTagFilterEnvVarName)
+				return generateDefaultFilters(clusterId)
+			}
+			filters[i] = &ec2.Filter{
+				Name:   aws.String(parts[0]),
+				Values: aws.StringSlice(strings.Split(parts[1], ",")),
+			}
+		}
+		return filters
+	}
+	return generateDefaultFilters(clusterId)
+}
+
+// Generate default EC2 filter for usage with ECs DescribeInstances call based on EC2 tags
+// of instance where Ingress Controller pod was started.
+func generateDefaultFilters(clusterId string) []*ec2.Filter {
+	return []*ec2.Filter{
+		{
+			Name:   aws.String("tag:" + clusterIDTagPrefix + clusterId),
```

Comment:
It should not have a `tag:` prefix IMO

Reply:
Not only should, but must have. You can check DescribeInstances documentation.

Reply:
Ok I see, this I didn't know.